src/tallies/tally.cpp

```diff
@@ -1271,4 +1272,20 @@ extern "C" size_t tallies_size()
   return model::tallies.size();
 }
 
+// given a tally ID, remove it from the tallies vector
+extern "C" int openmc_remove_tally(int32_t index)
+{
+  // check that id is in the map
+  if (index < 0 || index > model::tallies.size()) {
+    return OPENMC_E_OUT_OF_BOUNDS;
+  }
+  // grab tally so it's ID can be obtained to remove the (ID,index) pair from tally_map
+  auto& tally = model::tallies[index];
+  model::tally_map.erase(tally->id_);
```

Comment:
I believe this should already happen in the `Tally` destructor during the call to `tallies.erase` below

Reply:
Nice catch, thanks!